packages/headless/src/api/platform-client.ts

```diff
@@ -69,46 +70,38 @@ export class PlatformClient {
           ...processedOptions.headers,
         },
         body: JSON.stringify(processedOptions.requestParams),
+        signal: options.signal,
       });
       if (isThrottled(response.status)) {
         throw response;
       }
       return response;
     };
 
-    try {
-      const response = await backOff(request, {
-        retry: (e: Response) => {
-          const shouldRetry = e && isThrottled(e.status);
-          shouldRetry && options.logger.info('Platform retrying request');
-          return shouldRetry;
-        },
-      });
-      if (response.status === 419) {
-        processedOptions.logger.info('Platform renewing token');
-        const accessToken = await processedOptions.renewAccessToken();
+    const response = await backOff(request, {
+      retry: (e: Response) => {
+        const shouldRetry = e && isThrottled(e.status);
+        shouldRetry && options.logger.info('Platform retrying request');
+        return shouldRetry;
+      },
+    });
+    if (response.status === 419) {
+      processedOptions.logger.info('Platform renewing token');
+      const accessToken = await processedOptions.renewAccessToken();
 
-        if (accessToken !== '') {
-          return PlatformClient.call({...processedOptions, accessToken});
-        }
+      if (accessToken !== '') {
+        return PlatformClient.call({...processedOptions, accessToken});
       }
+    }
 
-      const body = (await response.json()) as ResponseType;
-
-      options.logger.info({response, body, requestInfo}, 'Platform response');
+    const body = (await response.json()) as ResponseType;
 
-      return {
-        response,
-        body,
-      };
-    } catch (error) {
-      options.logger.error({error, requestInfo}, 'Platform error');
+    options.logger.info({response, body, requestInfo}, 'Platform response');
 
-      return {
-        response: error,
-        body: (await error.json()) as ResponseType,
```

Comment:
I don't really understand/remember why an error was caught here and returned as a response...
You can't call .json on an Error object, so it just throws a different error 😄

Reply:
https://developer.mozilla.org/en-US/docs/Web/API/Fetch_API/Using_Fetch
> The Promise returned from fetch() won’t reject on HTTP error status even if the response is an HTTP 404 or 500. Instead, it will resolve normally (with ok status set to false), and it will only reject on network failure or if anything prevented the request from completing.

I think we should let it throw? And it will be handled by the error logger. Right now it throws when being aborted which is expected.

Reply:
In the tests this one fails:
`should not throw and return a response when backOff returns a rejected promise`
Why would backoff return a rejected promise? If fetch doesn't throw on a `429` status?

Reply:
I think backoff would return a rejected promise if it reaches some limit, like, after X amount of retries (I don't know the default value for the amount of retry).

So fetch itself would not throw, but if backoff retries a 100 times and always gets a `429`, we'd get a rejection.

That would be my assumption 🤔

Reply:
As for just letting it throw, remember that any platform call error, we want to also "bubble them up" all the way to the state, so that users could leverage the platform error in a component (eg: to show query errors).

For example: 

Too many retries (`429`)/invalid credentials ( search token) / invalid query / platform unavailable /org id non existant etc.. 

These would all be the type of errors that we want to bubble up in the state (under search.error) so that you can leverage it in your app as an end user.

For cancelled queries, we don't want to bubble them up to the state, it should just log in the console.

As long as this is respected, feel free to modify this function as much as you need I guess :)

Reply:
Right, the backoff throws the response when it's over the limit, I've handled that case, returning the response when the error obj has a `body`, but throwing if there isn't.

Reply:
Yes, the backOff function will throw when the attempt limit is reached (`10` is the default) or if the `retry` function returns `false` before the limit is hit.